src/java_bytecode/java_entry_point.cpp

```diff
@@ -123,13 +170,19 @@ bool java_static_lifetime_init(
       code_function_callt function_call;
       function_call.lhs()=nil_exprt();
       function_call.function()=it->second.symbol_expr();
+      function_call.add_source_location()=source_location;
       code_block.add(function_call);
     }
   }
 
   return false;
 }
 
+static bool is_string_array(const typet& t)
```

Comment:
removed `is_string_array` and `cast_if_necessary`functions, each was only used once
`is_constructor` and `is_virtual` are now a class methods